### PR TITLE
Hotfix/calculate-number-to-invite

### PR DESCRIPTION
**Pull request by fais-2**
## Type of changes

- [ ] Refactoring (non-breaking change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [ ] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [ ] I have followed the code style of the project
- [ ] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [ ] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: terraform/src/calculateNumberToInviteLambda/calculateNumberToInviteLambda.js [hunk 1]
```diff
@@ -175,6 +183,15 @@ export async function getItemsFromTable(table, client, key) {
   return response;
 }
 
+/**
+ * Invokes a participant list with the provided payload.
+ * @async
+ * @function invokeParticipantListLambda
+ * @param {string} lambdaName - The name of the participantList.
```
Comment (chenglawson1) [inline terraform/src/calculateNumberToInviteLambda/calculateNumberToInviteLambda.js:190]: This parameter is the name of the lambda to be invoked, not the name of the participant list.

File: terraform/src/calculateNumberToInviteLambda/calculateNumberToInviteLambda.js [hunk 2]
```diff
@@ -175,6 +183,15 @@ export async function getItemsFromTable(table, client, key) {
   return response;
 }
 
+/**
+ * Invokes a participant list with the provided payload.
```
Comment (chenglawson1) [inline terraform/src/calculateNumberToInviteLambda/calculateNumberToInviteLambda.js:187]: Invokes participant list lambda.

File: terraform/src/calculateNumberToInviteLambda/calculateNumberToInviteLambda.js [hunk 3]
```diff
@@ -196,6 +213,15 @@ export async function invokeParticipantListLambda(
...
 // Remove them from the object
 // Store the personId in selectedParticipants set
+
+/**
+ * @function getParticipantsInQuintile
+ * @param {Array} quintilePopulation - An array of objects representing the quintile population.
+ * @param {number} quintileTarget - The target number of participants to select.
+ * @param {number} nationalForecastUptake - The national forecast uptake value.
+ * @param {string} Q - A string representing the quintile block.
+ * @returns {Array} An Array containing the selected participants.
```
Comment (chenglawson1) [inline terraform/src/calculateNumberToInviteLambda/calculateNumberToInviteLambda.js:223]: Return type is a Set (not Array) or Error.

File: terraform/src/calculateNumberToInviteLambda/calculateNumberToInviteLambda.js [hunk 4]
```diff
@@ -270,6 +296,16 @@ export const getParticipantsInQuintile = (
...
 };
 
+/**
+ * Generates a block of items from the participantList within the specified bounds for a given quintile.
+ *
+ * @function generateQuintileBlocks
+ * @param {Array} participantList - The list of items to generate the quintile block from.
+ * @param {number} lowerBound - The lower bound index for slicing the list.
+ * @param {number} upperBound - The upper bound index for slicing the list.
+ * @param {number} quintile - The quintile number for which the block is generated.
```
Comment (chenglawson1) [inline terraform/src/calculateNumberToInviteLambda/calculateNumberToInviteLambda.js:306]: Please check again, I think this parameter is a string.

**Review by chenglawson1 — APPROVED** [review]

**Review by Amar-NHS — APPROVED** [review]

**Pull request merged**